subprojects/core-api/src/main/java/org/gradle/api/artifacts/result/ResolutionResult.java

```diff
@@ -89,4 +91,14 @@
      */
     void allComponents(Closure closure);
 
+    /**
+     * The attributes that were requested. Those are the attributes which
+     * are used during variant aware resolution, to select the variants.
+     * Attributes returned by this method are <i>desugared</i>, meaning that
+     * they have lost their rich types and can only be of type Boolean or String.
```

Comment:
From what I read in `AttributeDesugaring`, not.

Reply:
Indeed so that means `TargetJvmVersion` which has type `Integer` desugars to a `String` but gets written to metadata as an `Integer`. While not strictly related to the PR here, is that not some inconsistency?

Reply:
There is some inconsistency. As to if it's a problem here I'm not sure.